kernel/src/process.rs

```diff
@@ -296,17 +297,17 @@ pub struct Process<'a> {
 
     /// Essentially a list of callbacks that want to call functions in the
     /// process.
-    tasks: RingBuffer<'a, Task>,
+    tasks: MapCell<RingBuffer<'a, Task>>,
 
     /// Name of the app. Public so that IPC can use it.
     pub package_name: &'static str,
 
     /// Values kept so that we can print useful debug messages when apps fault.
-    debug: ProcessDebug,
+    debug: MapCell<ProcessDebug>,
```

Comment:
Why make this a MapCell as opposed to leaving the fields as `Cell`s? This seems OK to, but was there any particular logic behind the decision?

Reply:
I thought that was more consistent with Tock, for example in capsules using a cell around an `App` rather than having every field in an `App` use cells.

Reply:
I think Grants are pretty different as they basically _need_ to be reified into mutable references in order to make them general enough, whereas as this can easily be just a bunch of `Cell`s (which is more consistent with how we write non-Grant capsule code generally).

However, this is a small and contained enough design decision that i don't think it's worth kvetching over. I think using a MapCell here is just fine, was just wondering if there was a strict need for it I was missing.